File: Data/Entities/Target.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace FoodDiary.Data.Entities
{
    public class Target
    {
        public int Id { get; set; }
        public User User { get; set; }        
        
        public bool relevance { get; set; }
        public DateTime DateStart { get; set; }

        public DateTime DateFinish { get; set; }
        public double TargetBodyWeight { get; set; }

        public DailyRate DailyRate { get; set; }
        
        [ForeignKey(nameof(DailyRate))]
        public int DailyRateId { get; set; }

    }
}

File: client/src/app/shared/User/Target.ts
import { DailyRate } from "./DailyRate";

export class Target {
    id: number = 0;
    relevance: boolean = false;
    dateStart: Date = new Date();
    dateFinish: Date = new Date();
    currentBodyWeight: number = 0;
    targetBodyWeight: number = 0;
    dailyRate: DailyRate = new DailyRate();
}